### PR TITLE
SDKCF-2842: [Bug][IAM][iOS] - getConfig call is blocking and the retry can occur on main thread

### PR DESCRIPTION
**Pull request by malayli**
# Description
RInAppMessaging is blocking the main thread.
Running the retry handler on the in app queue solves this issue.

Steps:
* disconnect network
* launch demoapp and go to analytics and tap track event
* put app in background
* connect to network
* bring app to foreground

## Links
https://jira.rakuten-it.com/jira/browse/SDKCF-2842

### CODE REVIEW

**Comment by malayli** [comment]
@donnie-jp can you review this PR?

**Review by donnie-jp — APPROVED** [review]
CI failed.
Comments inline but overall LGTM.

File: RInAppMessaging/Classes/RInAppMessaging.swift [hunk 1]
```diff
@@ -47,7 +46,7 @@
     static func configure(dependencyManager: DependencyManager) {
         self.dependencyManager = dependencyManager
 
-        inAppQueue.async(flags: .barrier) {
+        dependencyManager.resolve(type: DispatchQueue.self)?.async(flags: .barrier) {
```
Comment (donnie-jp) [inline RInAppMessaging/Classes/RInAppMessaging.swift]: Can we prevent duplicating this verbose expression?
Maybe a computed property `inAppQueue` ?

Comment (malayli) [reply RInAppMessaging/Classes/RInAppMessaging.swift]: Done.

File: RInAppMessaging/Classes/RInAppMessaging.swift [hunk 2]
```diff
@@ -11,9 +11,11 @@
 /// Conforms to NSObject and exposed with objc tag to make it work with Obj-c projects.
 @objc public class RInAppMessaging: NSObject {
 
-    private static let inAppQueue = DispatchQueue(label: "IAM.Main", qos: .utility, attributes: [])
     internal private(set) static var initializedModule: InAppMessagingModule?
     private(set) static var dependencyManager: DependencyManager?
+    private static var inAppQueue: DispatchQueue? {
+        dependencyManager?.resolve(type: DispatchQueue.self)
```
Comment (donnie-jp) [inline RInAppMessaging/Classes/RInAppMessaging.swift:17]: Would it be useful to have a guard here to catch this being nil and log an error?

**Pull request merged**